odoo/addons/test_performance/tests/test_performance.py

```diff
@@ -610,3 +613,75 @@ def test_relational_mapped(self):
         with self.assertQueryCount(3):
             for rec in recs:
                 rec.line_ids.mapped('value')
+
+
+@tagged('increment_perf')
+class TestIncrementFieldsSkipLock(TransactionCase):
+    """ Test the behavior of the function `increment_fields_skiplock`.
+
+    Note that, per-design, the function will not always update the requested
+    records in case of a (table/row-level) lock. This is reflected in these
+    tests as we make sure to check the integrity of the results whether
+    any record was updated or not.
+    """
+
+    def setUp(self):
```

Comment:
Would it be possible to instead extend "SavepointCase" and use a setupClass instead of setUp ?

Small detail but we gain a few queries as data will only be created once.

Reply:
It should work with the "base" `TransactionCase` as well (https://www.odoo.com/documentation/master/developer/reference/backend/testing.html#odoo.tests.common.TransactionCase).
And it's done!
Thanks!